File: src/main/java/com/example/chatService/component/WebSocketEventPublisher.java
package com.example.chatService.component;

import com.example.chatService.dto.PublishAcceptFriendEvent;
import com.example.chatService.dto.PublishFriendEvent;
import lombok.RequiredArgsConstructor;
import org.springframework.messaging.simp.SimpMessagingTemplate;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
@RequiredArgsConstructor
public class WebSocketEventPublisher {

    private final SimpMessagingTemplate messagingTemplate;

    public void publishFriendEvent(String userId, PublishFriendEvent publishFriendEvent) {
        messagingTemplate.convertAndSend("/topic/friends/" + userId, publishFriendEvent);
    }

    public void publishAcceptFriendEvent(String userId, PublishAcceptFriendEvent publishFriendEvent) {
        messagingTemplate.convertAndSend("/topic/friends/" + userId, publishFriendEvent);
    }
}

File: src/main/java/com/example/chatService/component/WebSocketEventPublisher.kt
package com.example.chatService.component

import com.example.chatService.dto.PublishAcceptFriendEvent
import com.example.chatService.dto.PublishFriendEvent
import org.springframework.messaging.simp.SimpMessagingTemplate
import org.springframework.stereotype.Component

@Component
class WebSocketEventPublisher(
    private val messagingTemplate: SimpMessagingTemplate,
) {
    fun publishFriendEvent(userId: String, publishFriendEvent: PublishFriendEvent) {
        messagingTemplate.convertAndSend("/topic/friends/$userId", publishFriendEvent)
    }

    fun publishAcceptFriendEvent(userId: String, publishFriendEvent: PublishAcceptFriendEvent) {
        messagingTemplate.convertAndSend("/topic/friends/$userId", publishFriendEvent)
    }
}
